### ISSUE TITLE
[Feature]: On Map / In Campaign image selection option

### ISSUE THREAD

**Issue by Jmr3366** [RPTools/maptool#3738]
### Feature Request

Include an option in the image selection dialog box where the "Default" folder structure is listed, include a method to select a map (or campaign file) where the images listed come from the selected map (or campaign file).
Current dialog
![image](https://user-images.githubusercontent.com/100969108/200417732-09f773f6-e52e-46c7-97ac-bd9157eefa59.png)

### The Solution you'd like

Something like this (purely as an example).  "Maps" would be a folder of current maps in the campaign file.
![image](https://user-images.githubusercontent.com/100969108/200419829-9807d1dd-c645-4adf-8020-7931b10cf691.png)

### Alternatives that you've considered.

You can already find an asset ID, locate the respective *. file in assetcache.  Copy, rename, and place the file in the appropriate place. 

This works but is not "friendly".  

It also create data duplication as renames of the file _could_ create another copy in assetcache.

### Additional Context

I put a "wish" of how I'd like to see the dialog box function.

**Comment by Azhrei** [comment]
> It also create data duplication as renames of the file could create another copy in assetcache.

No, the name of the asset image in the assetcache is its checksum.  If the checksum doesn't change, ie. the content of the image doesn't change, then it's a single asset and exists in the campaign only once.

----

I love this idea overall!

So this would be sort of like how PDFs are containers of images; there would be a folder to represent a campaign file, then subfolders to represent maps, and subfolders to represent tokens.  The assets used then appear as the leaves of the tree.  Campaigns, exported maps, and saved tokens would all appear as folders in the Asset Library panel...

Each folder/subfolder could have a "meta folder" named "(ALL)" or something that holds assets from all entries in that folder and its subfolders.  Each folder would need to show the thumbnail embedded in the campaign, map, or token, as its image in the mouseover tooltip (perhaps with the number of assets it contains).

This would also integrate the idea of dragging a map onto the map view to import that map into the campaign, the same way that tokens can be dragged now.

This is very cool.  I can see lots of uses for it — reusing token images, but also textures.  And the user doesn't have to search all over the system trying to find the exact same image.

----

To take the idea one step further, metadata from the various subfolders could be imported by drag/drop as well.  If you're creating a new campaign and want to import some or all of the token type properties from an old one, there could be another meta folder (named "(PROPERTIES)"?) where token property types, sight types, light types, and other settings could be dragged to the current map to change the settings for the currently loaded campaign.
